src/instance.rs

```diff
@@ -709,9 +720,22 @@ impl<T> Py<T> {
     /// of the pointer or decrease the reference count (e.g. with [`pyo3::ffi::Py_DecRef`](crate::ffi::Py_DecRef)).
     #[inline]
     pub fn into_ptr(self) -> *mut ffi::PyObject {
-        let ptr = self.0.as_ptr();
-        std::mem::forget(self);
-        ptr
+        ManuallyDrop::new(self).0.as_ptr()
```

Comment:
This is a nice idiom of using `ManuallyDrop` for conciseness instead of safety, compared to using `forget`!

Reply:
Yes, it only works when pulling a `Copy` field out of the `ManuallyDrop`, but fortunately that's what we're doing here! 😂